api/src/scripts/configure_preset_datasources.ts

```diff
@@ -0,0 +1,128 @@
+import { DataSource as Source } from 'typeorm';
+import { Type } from 'class-transformer';
+import { IsInt, IsString, ValidateNested } from 'class-validator';
+import path from 'path';
+import { Like } from 'typeorm';
+import { DataSourceConfig } from '../api_models/datasource';
+import { dashboardDataSource } from '../data_sources/dashboard';
+import DataSource from '../models/datasource';
+import { PRESET_PREFIX } from '../utils/constants';
+import { configureDatabaseSource } from '../utils/helpers';
+import { maybeEncryptPassword } from '../utils/encryption';
+import { validate } from '../middleware/validation';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require('dotenv').config({ path: path.join(__dirname, '../../.env') });
+
+class DatabaseSource {
+  @IsString()
+  key: string;
+
+  @Type(() => DataSourceConfig)
+  config: DataSourceConfig;
+}
+
+class BaseConfig {
+  @IsString()
+  host: string;
+}
+
+class DatabaseConfig extends BaseConfig {
+  @IsInt()
+  port: number;
+
+  @IsString()
+  username: string;
+
+  @IsString()
+  password: string;
+
+  @IsString()
+  database: string;
+}
+
+class PresetDatasources {
+  @ValidateNested({ each: true })
+  @Type(() => DatabaseSource)
+  postgresql: DatabaseSource[];
+
+  @ValidateNested({ each: true })
+  @Type(() => DatabaseSource)
+  mysql: DatabaseSource[];
+
+  @ValidateNested({ each: true })
+  @Type(() => DatabaseSource)
+  http: DatabaseSource[];
+}
+
+async function upsert() {
+  console.log('Starting upsert of preset datasources');
+  if (!dashboardDataSource.isInitialized) {
+    await dashboardDataSource.initialize();
+  }
+  const queryRunner = dashboardDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const data: PresetDatasources = validate(PresetDatasources, require('../data_sources/preset.json'));
```

Comment:
I think we may need to make sure that all preset data sources used in preset dashboards are declared here since the preset data sources JSON is manually edited

Reply:
Done